src/measure_theory/group/prod.lean

```diff
@@ -171,57 +175,95 @@ lemma measure_mul_right_ne_zero [is_mul_left_invariant μ] {E : set G}
-  `0 < ν(Ex⁻¹) < ∞`. The first inequality follows from §59, Th. D, but I couldn't find the second
-  inequality. For this reason, we use a compact `E` instead of a measurable `E` as in [Halmos], and
-  additionally assume that `ν` is a regular measure (we only need that it is finite on compact
-  sets). -/
 @[to_additive]
-lemma measure_lintegral_div_measure [t2_space G] [is_mul_left_invariant μ]
-  [is_mul_left_invariant ν] [regular ν] {E : set G} (hE : is_compact E) (h2E : ν E ≠ 0)
-  (f : G → ℝ≥0∞) (hf : measurable f) :
-  μ E * ∫⁻ y, f y⁻¹ / ν ((λ h, h * y⁻¹) ⁻¹' E) ∂ν = ∫⁻ x, f x ∂μ :=
+lemma measure_mul_lintegral_eq [is_mul_left_invariant μ]
+  [is_mul_left_invariant ν] {E : set G} (Em : measurable_set E) (f : G → ℝ≥0∞) (hf : measurable f) :
+  μ E * ∫⁻ y, f y ∂ν = ∫⁻ x, ν ((λ z, z * x) ⁻¹' E) * f (x⁻¹) ∂μ :=
 begin
-  have Em := hE.measurable_set,
-  symmetry,
-  set g := λ y, f y⁻¹ / ν ((λ h, h * y⁻¹) ⁻¹' E),
-  have hg : measurable g := (hf.comp measurable_inv).div
-    ((measurable_measure_mul_right ν Em).comp measurable_inv),
   rw [← set_lintegral_one, ← lintegral_indicator _ Em,
-    ← lintegral_lintegral_mul (measurable_const.indicator Em).ae_measurable hg.ae_measurable,
+    ← lintegral_lintegral_mul (measurable_const.indicator Em).ae_measurable hf.ae_measurable,
     ← lintegral_lintegral_mul_inv μ ν],
   swap, { exact (((measurable_const.indicator Em).comp measurable_fst).mul
-      (hg.comp measurable_snd)).ae_measurable },
+      (hf.comp measurable_snd)).ae_measurable },
   have mE : ∀ x : G, measurable (λ y, ((λ z, z * x) ⁻¹' E).indicator (λ z, (1 : ℝ≥0∞)) y) :=
   λ x, measurable_const.indicator (measurable_mul_const _ Em),
   have : ∀ x y, E.indicator (λ (z : G), (1 : ℝ≥0∞)) (y * x) =
     ((λ z, z * x) ⁻¹' E).indicator (λ (b : G), 1) y,
   { intros x y, symmetry, convert indicator_comp_right (λ y, y * x), ext1 z, refl },
-  have h3E : ∀ y, ν ((λ x, x * y) ⁻¹' E) ≠ ∞ :=
-    λ y, (is_compact.measure_lt_top $ (homeomorph.mul_right _).compact_preimage.mpr hE).ne,
   simp_rw [this, lintegral_mul_const _ (mE _), lintegral_indicator _ (measurable_mul_const _ Em),
-    set_lintegral_one, g, inv_inv,
-    ennreal.mul_div_cancel' (measure_mul_right_ne_zero ν h2E _) (h3E _)]
+    set_lintegral_one],
 end
 
-/-- This is roughly the uniqueness (up to a scalar) of left invariant Borel measures on a second
-  countable locally compact group. The uniqueness of Haar measure is proven from this in
-  `measure_theory.measure.haar_measure_unique` -/
 @[to_additive]
-lemma measure_mul_measure_eq [t2_space G] [is_mul_left_invariant μ]
-  [is_mul_left_invariant ν] [regular ν] {E F : set G}
-  (hE : is_compact E) (hF : measurable_set F) (h2E : ν E ≠ 0) : μ E * ν F = ν E * μ F :=
+lemma ae_measure_preimage_mul_right_ne_top [is_mul_left_invariant μ] [is_mul_left_invariant ν]
+  {E : set G} (Em : measurable_set E) (hμE : μ E < ∞) :
+  ∀ᵐ x ∂μ, ν ((λ y, y * x) ⁻¹' E) < ∞ :=
```

Comment:
2 things:
- the name should use `lt_top` instead of `ne_top`, given the result
- I think our convention is to use `≠ ∞` instead of `< ∞` in hypotheses. Several other lemmas in this PR should be modified in this way.
```suggestion
lemma ae_measure_preimage_mul_right_lt_top [is_mul_left_invariant μ] [is_mul_left_invariant ν]
  {E : set G} (Em : measurable_set E) (hμE : μ E ≠ ∞) :
  ∀ᵐ x ∂μ, ν ((λ y, y * x) ⁻¹' E) < ∞ :=
```

Reply:
I forgot the conventions (I'm not sure if they're written down anywhere. Fixed now.